tests/unittest/test_breakpad_resource.py

```diff
@@ -151,6 +151,18 @@ def test_extract_payload_json(self, request_generator):
         expected_dumps = {"upload_file_minidump": b"abcd1234"}
         assert bsp.extract_payload(req) == (expected_raw_crash, expected_dumps)
 
+    def test_extract_payload_bad_json(self, request_generator):
+        # If the JSON doesn't parse (invalid control character), it raises
+        # a MalformedCrashReport
+        data, headers = multipart_encode({"extra": '{"ProductName":"Firefox\n"}'})
```

Comment:
This has a `\n` but that's converted to a carriage return during Python parsing and carriage return is an invalid character in a JSON string.